linera-views/src/storage_metric.rs

```diff
@@ -0,0 +1,287 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    batch::{Batch, WriteOperation},
+    common::{KeyIterable, KeyValueIterable, KeyValueStoreClient},
+    memory::{create_memory_client, MemoryClient},
+};
+use async_lock::RwLock;
+use async_trait::async_trait;
+use std::sync::Arc;
+
+/// Array containing metric data for this client
+/// This can be used for storage fees or for other
+/// benchmarking purposes.
+#[derive(Clone, Copy, Default, Debug, Eq, PartialEq)]
+pub struct MetricStat {
+    /// The total number of keys read in calls to `read_key` and `read_multi_key`
+    pub n_reads: usize,
+    /// The number of missed cases in read_key and read_multi_key
+    pub n_miss_reads: usize,
+    /// The number of Put in the batches
+    pub n_puts: usize,
+    /// The total number of Delete in the batches
+    pub n_deletes: usize,
+    /// The total number of DeletePrefix in the batches
+    pub n_delete_prefix: usize,
+    /// The total data that went into the Batches
+    pub size_writes: usize,
+    /// The total data being read from
+    pub size_reads: usize,
+}
+
+/// The `MetricKeyValueClient` encapsulates a client and creates a new one that
+/// measures the operations being done
+#[derive(Clone)]
+pub struct MetricKeyValueClient<K> {
+    /// The inner client that is called by the metric one
+    pub client: K,
+    /// The data contained in the running of this container
+    pub metric_stat: Arc<RwLock<MetricStat>>,
```

Comment:
We may want to use [atomic counters](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU32.html#method.fetch_add) and avoid the lock. (And if we have to use a lock, we may not need an async one)

Reply:
Thank you, and thank you for pointing to the exact relevant entry of `fetch_add`.